modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1439,6 +1453,25 @@ def isin_func(df, values):
     _isfinite = Map.register(
         lambda df, *args, **kwargs: pandas.DataFrame(np.isfinite(df))
     )
+    _isinf = Map.register(  # Needed for numpy API
+        lambda df, *args, **kwargs: pandas.DataFrame(np.isinf(df))
+    )
+    _isnat = Map.register(  # Needed for numpy API
+        lambda df, *args, **kwargs: pandas.DataFrame(np.isnat(df))
+    )
+    _isneginf = Map.register(  # Needed for numpy API
+        lambda df, *args, **kwargs: pandas.DataFrame(np.isneginf(df))
+    )
+    _isposinf = Map.register(  # Needed for numpy API
+        lambda df, *args, **kwargs: pandas.DataFrame(np.isposinf(df))
+    )
+    _iscomplex = Map.register(  # Needed for numpy API
+        lambda df, *args, **kwargs: pandas.DataFrame(np.iscomplex(df))
+    )
+    _isreal = Map.register(  # Needed for numpy API
```

Comment:
Shouldn't we hint the dtypes here as well?

Reply:
If a `where` array is specified
> This condition is broadcast over the input. At locations where the condition is True, the out array will be set to the ufunc result. Elsewhere, the out array will retain its original value. This means there might be non-boolean values in the result. I'll add a comment to that effect. 

An example:
```
>>> out = numpy.array([2.0,2.0])
>>> numpy.isfinite(numpy.array([1.0,1.0]), where=[False, True], out=out)
array([2., 1.])
```